source/MRViewer/MRRibbonSchema.cpp

```diff
@@ -301,6 +302,17 @@ std::vector<RibbonSchemaHolder::SearchResult> RibbonSchemaHolder::search( const
         }
     }
 
+    // sort by available - unavailable tools
+    const auto& selectedObjs = SceneCache::getAllObjects<const Object, ObjectSelectivityType::Selected>();
+    std::sort( rawResult.begin(), rawResult.end(), [&selectedObjs] ( const auto& a, const auto& b )
+    {
+        if ( !a.first.item->item->isAvailable( selectedObjs ).empty() &&
+            b.first.item->item->isAvailable( selectedObjs ).empty() )
+            return false;
+        else
+            return true;
```

Comment:
we probably should use common requirerments function from menu

Reply:
reworked